src/http_microhttpd.c

```diff
@@ -759,9 +833,17 @@ static int get_query(struct MHD_Connection *connection, char **query)
 			continue;
 		}
 		strncpy(*query + j, elements[i], length - j);
+		if (i == 0) {
+			strcpy(query_str, "?");
+		} else {
+			strcat(query_str, "&");
+		}
+		strcat(query_str, *query);
```

Comment:
This (strcpy, strcat, strcat) can overflow when the input this too long.
Please use strncpy and strncat.

Reply:
Fixed

Reply:
And tested.

Reply:
Maybe just a wee tweak still needed......

Reply:
Really fixed :-D